snuba/datasets/processors/search_issues_processor.py

```diff
@@ -99,6 +127,43 @@ def _process_user(
 
         return
 
+    def _process_tags(
+        self, event_data: IssueEventData, processed: MutableMapping[str, Any]
+    ) -> None:
+        tags_maybe = event_data.get("tags", None)
+        if not tags_maybe:
+            processed["tags.key"], processed["tags.value"] = [], []
```

Comment:
I think you should `return` here.

Reply:
I actually had it return early before, but then the extra params won't get processed:
```
processed["release"] = promoted_tags.get(
            "sentry:release",
            event_data.get("release"),
        )
```

So events like:
```
{
        "project_id": 1,
        "organization_id": 2,
        "group_id": 3,
        "event_id": str(uuid.uuid4()),
        "retention_days": 90,
        "primary_hash": str(uuid.uuid4()),
        "datetime": datetime.utcnow().isoformat() + "Z",
        "platform": "other",
        "data": {
            "tags": {}
            "release": "release@1.2.3"
        }
        ...
    }
```
would get skipped entirely.

Reply:
Ahh, right. I rearranged `process_tags` to be a bit more readable and to avoid doing dupe work.